### PR TITLE
Use LD_LIBRARY_PATH over RPATH for build dir .so

### PR DESCRIPTION
**Pull request by ocaisa**

### CODE REVIEW

File: easybuild/easyblocks/f/flexiblas.py [hunk 1]
```diff
@@ -104,20 +103,21 @@ def configure_step(self):
...
 
     def test_step(self):
         """Run tests using each of the backends."""
 
+        additional_pretestopts = ''
+        if build_option('rpath'):
+            # inject build location of libflexiblas.so.3,
+            # to ensure that test binaries can find the FlexiBLAS library
+            build_libdir = os.path.join(self.builddir, 'easybuild_obj', 'lib')
+            additional_pretestopts = "LD_LIBRARY_PATH=%s:$LD_LIBRARY_PATH" % build_libdir
```
Comment (boegel) [inline easybuild/easyblocks/f/flexiblas.py:116]: `$LD_LIBRARY_PATH` is ignored when RPATH is involved, so this is pointless?

I'm working on a similar change though: only injecting `-Wl,-rpath=...` into `$FFLAGS` & co for the test command  (via `testopts`), testing that now...

Comment (ocaisa) [reply easybuild/easyblocks/f/flexiblas.py:116]: No not pointless, this is _only_ required for tests, once installed the RPATH is already set up

Comment (ocaisa) [reply easybuild/easyblocks/f/flexiblas.py:116]: For POWER, may need to include `lib64`?

Comment (boegel) [reply easybuild/easyblocks/f/flexiblas.py:116]: No, it's always `lib` I think, regardless of CPU architecture. Depending on the OS, you could get `lib64` though (on SLES for example), so perhaps it's better to play safe and inject both `lib` and `lib64`.

W.r.t. the relation between `$LD_LIBRARY_PATH` and RPATH, `$LD_LIBRARY_PATH` is indeed still used as a fallback for libraries that were not found via the paths included in the RPATH section of the binary, so it's the right approach here I think.

**Comment by boegel** [comment]
fixed slightly differently in f8c4ed184e6177b0dafaa5ae433d7dfa647ff0d9, which is now included in https://github.com/easybuilders/easybuild-easyblocks/pull/2369

**Pull request closed without merging**